python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
+        properties: Optional[Properties] = None,
+    ) -> RestTable:
+        namespace_and_table = self._split_namespace_and_table(identifier)
+        properties = properties or {}
+        request = CreateTableRequest(
+            name=namespace_and_table["table"],
+            location=location,
+            table_schema=schema,
+            partition_spec=partition_spec,
+            write_order=sort_order,
+            properties=properties,
+        )
+        serialized_json = request.json()
+        response = requests.post(
+            self.url(Endpoints.create_table, namespace=namespace_and_table["namespace"]),
+            data=serialized_json,
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {409: TableAlreadyExistsError})
+
+        return RestTable(identifier=identifier, **response.json())
+
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
+        namespace_concat = NAMESPACE_SEPARATOR.join(self.identifier_to_tuple(namespace or ""))
+        response = requests.get(
+            self.url(Endpoints.list_tables, namespace=namespace_concat),
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
+        response = ListTablesResponse(**response.json())
+
+        return [(*entry.namespace, entry.name) for entry in response.identifiers]
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        response = requests.get(
+            self.url(Endpoints.load_table, prefixed=True, **self._split_namespace_and_table(identifier)), headers=self.headers
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
+        return Table(identifier=identifier, **response.json())
+
+    def drop_table(self, identifier: Union[str, Identifier], purge_requested: bool = False) -> None:
+        response = requests.delete(
+            self.url(Endpoints.drop_table, prefixed=True, purge=purge_requested, **self._split_namespace_and_table(identifier)),
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        raise NotImplementedError("Not implemented")
```

Comment:
I think this should call `self.drop_table(identifier, purge_requested=True)`.

Reply:
I'll update that in the code. I was assuming that it would only truncate the table.